producer-c/producer-cloudwatch-integ/CanaryStreamCallbacks.cpp

```diff
@@ -0,0 +1,216 @@
+
+    DLOGE("CanaryStreamErrorReportHandler got error %lu at time %" PRIu64 " for stream % " PRIu64 " for upload handle %" PRIu64, statusCode, erroredTimecode, streamHandle, uploadHandle);
+    pCanaryStreamCallbacks->streamErrorDatum.SetValue(1.0);
+    pCanaryStreamCallbacks->streamErrorDatum.SetUnit(Aws::CloudWatch::Model::StandardUnit::None);
+    CanaryStreamSendMetrics(pCanaryStreamCallbacks, pCanaryStreamCallbacks->streamErrorDatum);
+
+    return STATUS_SUCCESS;
+}
+
+STATUS CanaryStreamFragmentAckHandler(UINT64 customData,
+                                      STREAM_HANDLE streamHandle,
+                                      UPLOAD_HANDLE uploadHandle,
+                                      PFragmentAck ack)
+{
+    PCanaryStreamCallbacks pCanaryStreamCallbacks = (PCanaryStreamCallbacks) customData;
+    UINT64 time = GETTIME();
+    UINT64 timeOfFragmentEndSent = pCanaryStreamCallbacks->timeOfNextKeyFrame->find(ack->timestamp)->second;
+    if (ack->ackType != FRAGMENT_ACK_TYPE_BUFFERING) {
+        DLOGD("ack type %u latency %" PRIu64, ack->ackType, (GETTIME() - timeOfFragmentEndSent) / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+    }
+    switch (ack->ackType) {
+        case FRAGMENT_ACK_TYPE_RECEIVED:
+            pCanaryStreamCallbacks->receivedAckDatum.SetValue((GETTIME() - timeOfFragmentEndSent) / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+            pCanaryStreamCallbacks->receivedAckDatum.SetUnit(Aws::CloudWatch::Model::StandardUnit::Milliseconds);
+            CanaryStreamSendMetrics(pCanaryStreamCallbacks, pCanaryStreamCallbacks->receivedAckDatum);
+            break;
+        case FRAGMENT_ACK_TYPE_PERSISTED:
+            pCanaryStreamCallbacks->persistedAckDatum.SetValue((GETTIME() - timeOfFragmentEndSent) / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+            pCanaryStreamCallbacks->persistedAckDatum.SetUnit(Aws::CloudWatch::Model::StandardUnit::Milliseconds);
+            CanaryStreamSendMetrics(pCanaryStreamCallbacks, pCanaryStreamCallbacks->persistedAckDatum);
+            pCanaryStreamCallbacks->timeOfNextKeyFrame->erase(ack->timestamp);
+            break;
+        case FRAGMENT_ACK_TYPE_ERROR:
+            DLOGE("Received Error Ack timestamp %" PRIu64 " fragment number %s error code %lu", ack->timestamp, ack->sequenceNumber, ack->result);
+            break;
+        default:
+            break;
+    }
+    DLOGD("CanaryStreamFragmentAckHandler took %" PRIu64, GETTIME() - time);
+    return STATUS_SUCCESS;
+}
+
+VOID OnPutMetricDataResponseReceivedHandler(const Aws::CloudWatch::CloudWatchClient* cwClient,
+                                            const Aws::CloudWatch::Model::PutMetricDataRequest& request,
+                                            const Aws::CloudWatch::Model::PutMetricDataOutcome& outcome,
+                                            const std::shared_ptr<const Aws::Client::AsyncCallerContext>& context) {
+    if (!outcome.IsSuccess())
```

Comment:
formatting is off. Should be 

if (!outcome.IsSuccess()) {
} else {
}

Reply:
Fixed.